sktime/transformations/panel/dictionary_based/_sfa.py

```diff
@@ -518,8 +617,8 @@ def _mft(self, series, transformed=None, stds=None):
             self.inverse_sqrt_win_size,
         )
 
-        # lower bounding
-        transformed[:, 1::2] = transformed[:, 1::2] * -1
+        if not self.lower_bounding:
+            transformed[:, 1::2] = transformed[:, 1::2] * -1  # lower bounding
```

Comment:
So in our Java version of BOSS/TDE we dont do this, but for the WEASEL version we do. Like this lower_bounding == False mimics WEASEL, while lower_bounding == True is BOSS/TDE im pretty sure.

Probably need more input on what would be best or how much an impact this has.
(Also applies to DFT)

Reply:
@MatthewMiddlehurst Oh, really cool! I will try to give the code a review. But it might take till the next week.

To answer this question:
SFA can be used as a dimensionality reduction technique for indexing and the distance on two SFA words lower bounds the Euclidean distance (there is an analogy for SAX).

The lower_bounding check comes from this background. There is also a division by the length of the subsequence somewhere in the code. 

When solely looking at the multiplication by -1, it doesn't make a difference for classification, I believe.

Reply:
@patrickzib Thanks, would be great if you could take a look. No rush on this.

There is also the inverse_sqrt_win_size affected by the setting. Individually it didnt make a lot of difference to TDE at least, but there were a few minor things that stacked up. Let me know if you want anything changed, SFA is quite widely used so I dont want to step on anyones feet :)

There are a couple of things im not so happy with i.e. the numerous if/else for different data types, but different functions are required for uncapped word_length and alphabet size unfortunetly.

Reply:
@MatthewMiddlehurst 

Hard to tell: Indexing is not the focus of sktime. Thus, it might be safe to comment these.

Have you tried to do a critical difference plot with and without these lines of code? That would be interesting.

Reply:
@patrickzib I did this, and the difference was minor for sure, I stopped TDE from doing it because it was very slightly worse and slightly slower. Any difference was definitely not significant from this alone.

Would be good to keep the functionality and sktime is more than just classification, I’m pretty sure the way it is set up now makes sense, but good to check.